File: app/src/main/java/com/booboot/vndbandroid/bean/vndbandroid/Vote.java
package com.booboot.vndbandroid.bean.vndbandroid;

import com.booboot.vndbandroid.R;

import java.text.DecimalFormat;
import java.util.regex.Pattern;

/**
 * Created by od on 15/03/2016.
 */
public class Vote {
    public final static String DEFAULT = "Add a vote";
    private final static DecimalFormat VOTE_FORMAT = new DecimalFormat("#.#");

    public static double outOf10(int vote) {
        return vote / 10.0;
    }

    public static String toString(int vote) {
        if (vote < 1) return DEFAULT;
        return toShortString(vote) + " (" + getName(outOf10(vote)) + ")";
    }

    public static String toShortString(int vote) {
        return VOTE_FORMAT.format(outOf10(vote));
    }

    public static boolean isValid(String vote) {
        if (vote == null) return false;
        Pattern pattern = Pattern.compile("[1-9](\\.[0-9])?");
        return pattern.matcher(vote).matches();
    }

    public static String getName(double vote) {
        if (vote >= 10) return "masterpiece";
        if (vote >= 9) return "excellent";
        if (vote >= 8) return "very good";
        if (vote >= 7) return "good";
        if (vote >= 6) return "decent";
        if (vote >= 5) return "so-so";
        if (vote >= 4) return "weak";
        if (vote >= 3) return "bad";
        if (vote >= 2) return "awful";
        if (vote >= 1) return "worst ever";
        if (vote >= 0) return "Other";
        return DEFAULT;
    }

    public static Integer getImage(double vote) {
        if (vote >= 8) return R.drawable.score_green;
        if (vote >= 7) return R.drawable.score_light_green;
        if (vote >= 6) return R.drawable.score_yellow;
        if (vote >= 4) return R.drawable.score_light_orange;
        if (vote >= 3) return R.drawable.score_orange;
        else return R.drawable.score_red;
    }
}

File: app/src/main/java/com/booboot/vndbandroid/model/vndbandroid/Vote.kt
package com.booboot.vndbandroid.model.vndbandroid

import androidx.annotation.ColorRes
import androidx.annotation.DrawableRes
import com.booboot.vndbandroid.App
import com.booboot.vndbandroid.R

import java.text.DecimalFormat
import java.util.regex.Pattern

object Vote {
    const val DEFAULT = "Add a vote"
    private val VOTE_FORMAT = DecimalFormat("#.#")

    fun outOf10(vote: Int?) = if (vote == null) 0f else vote / 10f

    fun toString(vote: Int) =
        if (vote < 1) DEFAULT else toShortString(vote) + " (" + getName(outOf10(vote)) + ")"

    fun toShortString(vote: Int?, defaultValue: String? = App.context.getString(R.string.dash)): String? =
        if (vote != null) VOTE_FORMAT.format(outOf10(vote)) else defaultValue

    fun isValid(vote: String?): Boolean {
        if (vote == null) return false
        val pattern = Pattern.compile("[1-9](\\.[0-9])?")
        return pattern.matcher(vote).matches()
    }

    fun getName(vote: Float) = when {
        vote >= 10 -> "masterpiece"
        vote >= 9 -> "excellent"
        vote >= 8 -> "very good"
        vote >= 7 -> "good"
        vote >= 6 -> "decent"
        vote >= 5 -> "so-so"
        vote >= 4 -> "weak"
        vote >= 3 -> "bad"
        vote >= 2 -> "awful"
        vote >= 1 -> "worst ever"
        vote >= 0 -> "Other"
        else -> DEFAULT
    }

    @ColorRes
    fun getColor(vote: Float?) = when {
        vote == null -> R.color.red
        vote >= 8 -> R.color.green
        vote >= 7 -> R.color.lightGreen
        vote >= 6 -> R.color.yellow
        vote >= 4 -> R.color.lightOrange
        vote >= 3 -> R.color.orange
        else -> R.color.red
    }

    @DrawableRes
    fun getDrawableColor(vote: Float?) = when {
        vote == null -> R.drawable.rounded_primary_background
        vote >= 8 -> R.drawable.rounded_green_background
        vote >= 7 -> R.drawable.rounded_light_green_background
        vote >= 6 -> R.drawable.rounded_yellow_background
        vote >= 4 -> R.drawable.rounded_light_orange_background
        vote >= 3 -> R.drawable.rounded_orange_background
        vote > 0 -> R.drawable.rounded_red_background
        else -> R.drawable.rounded_primary_background
    }

    @DrawableRes
    fun getDrawableColor10(vote: Int?) = getDrawableColor(outOf10(vote))
}